src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -273,9 +281,24 @@ private void OnSymbolEnd(SymbolAnalysisContext symbolEndContext, bool hasInvalid
                                 // Compute the set of candidate symbols referenced in all the documentation comments within the named type declarations.
                                 // This set is computed once and used for all the iterations of the loop.
                                 symbolsReferencedInDocComments = GetCandidateSymbolsReferencedInDocComments(namedType, symbolEndContext.Compilation, symbolEndContext.CancellationToken);
+
+                                // Compute the set of string arguments to DebuggerDisplay attributes applied to any symbol within the named type declaration.
+                                // These strings may have an embedded reference to the symbol.
+                                // This set is computed once and used for all the iterations of the loop.
+                                debuggerDisplayAttributeArguments = GetDebuggerDisplayAttributeArguments(namedType);
+
                                 first = false;
                             }
 
+                            // Simple heuristic for members referenced in DebuggerDisplayAttribute's string argument:
+                            // bail out if any of the DebuggerDisplay string arguments contains the member name.
+                            // In future, we can consider improving this heuristic to parse the embedded expression
+                            // and resolve symbol references.
+                            if (debuggerDisplayAttributeArguments.Any(arg => arg.Contains(member.Name)))
```

Comment:
1. use of linq is generally discouraged during analysis phase.  That said, there aren't likely to be many hits here.  So it's possibly ok.  
2. this likely won't be ok for VB (case insensitive).  but maybe that's too niche to care aobut.